src/lib/map.svelte

```diff
@@ -1,39 +1,67 @@
 <script lang="ts">
 	import mapboxgl, { Map } from 'mapbox-gl';
 	import { onMount, onDestroy } from 'svelte';
-	import type { Web3EnrichedMapboxFeature } from '../types';
+	import type { Web3EnrichedMapboxFeature, metadata } from '../types';
 
 	let map: Map;
 
-	function createPopupContent(feature: Web3EnrichedMapboxFeature): string {
+	async function getPopupMetadata(cid: string): Promise<metadata | undefined> {
+		const requestOptions: RequestInit = {
+			method: 'GET',
+			redirect: 'follow' as RequestRedirect
+		};
+
+		try {
+			const response = await fetch(`http://localhost:7777/api/metadata/${cid}`, requestOptions);
+			if (!response.ok) {
+				throw new Error(`Error fetching metadata for CID ${cid}: ${response.statusText}`);
+			}
+			const data: metadata = await response.json();
+			return data;
+		} catch (error) {
+			console.error(`Failed to fetch metadata for CID ${cid}:`, error);
+			return undefined;
+		}
+	}
+
+	async function createPopupContent(feature: Web3EnrichedMapboxFeature): Promise<string> {
 		const properties = feature.properties;
+		const metadata = await getPopupMetadata(properties.cid);
+		if (!metadata) {
+			console.warn(`No metadata found for CID ${properties.cid}.`);
+		}
+
 		return `
 		<b>Popup Title</b><br>
-		<span class="cid-text">CID: QmPK1s3pNYLi9ERiq3BDxKa4XosgWwFRQUydHUtz4YgpqB</span><br>
+		<span class="cid-text">CID: ${properties.cid}</span><br>
 		Row: ${properties.ROW}<br>
-		Path ${properties.PATH}<br>
-		Date acquired: July 26th, 2023<br>
-		Pinned on 5 IPFS nodes<br>
-		Stored in 3 Filecoin deals<br>
-		Available on S3: Yes ✅<br>
-		2 unsealed copies available<br>
+		Path: ${properties.PATH}<br>
+		Date acquired: ${new Date(properties.datetime).toLocaleDateString('en-US', {
+			year: 'numeric',
+			month: 'long',
+			day: 'numeric'
+		})}<br>
+
+		Pinned on ${metadata?.ipfs ?? 'N/A'} IPFS nodes<br> <!-- Example of including metadata -->
```

Comment:
I am making the assumption that there was a failure in the metadata call if it shows 'N/A'

Otherwise, it will show a value.

Just wanted to make sure that made sense. If there are no nodes pinning the data or no active storage deals, the API will return 0, right?